include/flash.h

```diff
@@ -137,27 +162,27 @@ static inline int flash_write_protection_set(struct device *dev, bool enable)
 	return api->write_protection(dev, enable);
 }
 
+#if defined(CONFIG_FLASH_PAGE_LAYOUT)
 /**
- *  @brief  Get layout of pages of flash.
- *
- *  Layout is provided in run-length format. For a flash memory which conststs
- *  uniform pages, layout will by simple the one couple of pages count and them
- *  size.
- *  For a flash memory which conststs pages of different sizes, layout wil be
- *  the sequence of such a couples.
+ *  @brief  Get size of flash page at certain flash offset.
  *
- *  @param  dev     :flash device
- *  @param  layout  :structure for store the layout
+ *  @param  dev: flash device
+ *  @param  offset: Offset within the page.
  *
- *  @return  0 on success, negative errno code on fail.
+ *  @return  Page size. It's 0 if page of the offset doesn't exist.
  */
-static inline int flash_page_layout(struct device *dev,
-				    struct pg_layout *layout)
-{
-	const struct flash_driver_api *api = dev->driver_api;
+size_t flash_get_page_size_by_offs(struct device *dev, off_t offset);
```

Comment:
we should be able to get also offset of the page - I don't think this is possible now, is it?

what about something like this:

struct flash_page_info {
    u32_t start_address;
    size_t size;
};
int flash_get_page_info_by_offset(struct device *dev, off_t offset, struct flash_page_info *pg);
int flash_get_page_info_by_index(struct device *dev, u32_t index, struct flash_page_info *pg);